index.js

```diff
@@ -0,0 +1,174 @@
+  })
+  .catch((error) =>{
+    console.log(error)
+    reportStatus(`Error: ${error.message}`)
+  })
+}
+
+const loadTrips = ()=>{
+  //create reference to all-trips element
+  let tripList = $(`#all-trips`);
+  //empty list before call to API
+  tripList.empty();
+  //display pending status to user
+  reportStatus('Loading trips....')
+  axios.get(URL)
+  .then((response) => {
+    response.data.forEach((trip) => {
+      tripList.append(`<li id="${trip.id}">${trip.name}</li>`)
+    });
+    reportStatus('Trips loaded!');
+  })
+  .catch((error) =>{
+    console.log(error)
+    reportStatus(`Error: ${error.message} `)
+  })
+}
+
+const createTrip = (event) =>{
+  event.preventDefault();
+
+  const newTripinfo = {
+    name: $(`input[name="name-create"]`).val(),
+    continent: $(`input[name="continent"]`).val(),
+    about: $(`input[name="about"]`).val(),
+    category: $(`input[name="category"]`).val(),
+    weeks: $(`input[name="weeks"]`).val(),
+    cost: $(`input[name="cost"]`).val()
+  }
+
+  reportStatus(`Attempting to create your new trip!`)
+  $(`input[name="name-create"]`).val('')
+  $(`input[name="continent"]`).val('')
+  $(`input[name="about"]`).val('')
+  $(`input[name="category"]`).val('')
+  $(`input[name="weeks"]`).val('')
+  $(`input[name="cost"]`).val('')
+
+  axios.post(URL,newTripinfo)
+  .then((response) => {
+    reportStatus('Your trip has been created')
+  })
+  .catch((error) =>{
+    reportStatus('fail')
+    if( error.message && error.response.data.errors){
+      reportError(error.message,error.response.data.errors)
+    }else
+    reportError(`encounted error: ${error.message}`)
+  })
+};
+
+
+const bookTrip = (event) =>{
+  event.preventDefault();
+
+  const tripInfo ={
+    name: $(`input[name="name"]`).val(),
+    email:$(`input[name="email"]`).val(),
+    tripID:$(`input[name="location-id"]`).val()
+  }
+
+  reportStatus(`Attempting to make your reservation`)
+  $(`input[name="name"]`).val('')
+  $(`input[name="email"]`).val('')
+  $(`input[name="location-id"]`).val('')
+
+  axios.post(reserveAtrip(tripInfo.tripID),tripInfo)
+  .then((response) => {
+    reportStatus('You trip has been booked')
+  })
+  .catch((error) => {
+    reportStatus('fail')
+    if( error.message && error.response.data.errors){
+      let v = error.response.data.errors
+      console.log(v)
+      reportError(error.message,error.response.data.errors)
+    }else
+    reportError(`encounted error: ${error.message}`)
+  })
+};
+
+
+$(document).ready(() => {
+  $(`#load`).click(loadTrips);
+  $(document).on("click","#all-trips",function(event){
+    loadTripdetails(event);
+  });
+
+  $(document).on("click","#make-res",function(){
+    $("#book-trip-form").toggle("slow");
+  });
+
+  $(`#book-trip-form`).submit(bookTrip);
+
+  $(document).on("click","#submit-button",function(event){
+    $(`#submit-button`).toggle();
+    $(`#book-trip-form`).append(`<button id="button-success" class="button success">Trip Booked!</button>`);
```

Comment:
This always has a successful button, even if the reservation wasn't made correctly!